akka-http/src/main/scala/akka/http/scaladsl/server/directives/FormFieldDirectives.scala

```diff
@@ -113,27 +129,28 @@ object FormFieldDirectives extends FormFieldDirectives {
   private def toMap(seq: Seq[(String, String)]): Map[String, String] = immutable.TreeMap(seq: _*)
 
   sealed trait FieldMagnet {
-    type Out
-    def apply(): Out
+    type U
+    def apply(): Directive[U]
```

Comment:
So, the typesafe fix is to add enough types here (and everywhere) to declare that the result of `pdf()` above actually is a `Directive` and not just anything.

That is we add the `Directive` type here and remove the `Directive` wrapping in all types below. Note that `type Directive1[T] = Directive[Tuple1[T]]` etc.